model_api/python/model_api/models/classification.py

```diff
@@ -354,12 +369,9 @@ def __init__(self, hierarchical_config) -> None:
         self.label_relations = hierarchical_config["label_tree_edges"]
         self.label_groups = hierarchical_config["cls_heads_info"]["all_groups"]
 
-    def _get_parent(self, label):
+        self.label_tree = SimpleLabelsGraph(list(self.label_to_idx.keys()))
         for child, parent in self.label_relations:
-            if label == child:
-                return parent
-
-        return None
+            self.label_tree.add_edge(parent, child)
 
     def resolve_labels(self, predictions):
```

Comment:
How aboug adding a docstring for explaining which format `predictions` should have?

Reply:
Added